## Análise de Correlação

In [4]:
# instalar versão atualizada
! sudo pip install scipy==1.11.4

     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 60.4/60.4 kB 3.6 MB/s eta 0:00:00
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 61.0/61.0 kB 4.8 MB/s eta 0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 35.8/35.8 MB 54.0 MB/s eta 0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 18.0/18.0 MB 82.2 MB/s eta 0:00:00
  Attempting uninstall: numpy
    Found existing installation: numpy 2.0.2
    Uninstalling numpy-2.0.2:
      Successfully uninstalled numpy-2.0.2
  Attempting uninstall: scipy
    Found existing installation: scipy 1.16.3
    Uninstalling scipy-1.16.3:
      Successfully uninstalled scipy-1.16.3
ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.
jax 0.7.2 requires numpy>=2.0, but you have numpy 1.26.4 which is incompatible.
jax 0.7.2 requires scipy>=1.13, but you have scipy 1.11.4 which is incompatible.
giddy 2.3.8 requires scipy>=1.12, but you have 

In [1]:
# instalar nova ferramenta para testes de hipoteses
! sudo pip install pingouin

In [2]:
# gerais
import numpy as np
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt

In [3]:
from scipy import stats as st
import pingouin as pg

In [4]:
import pandas as pd

arquivos = [
    'vendas_linha_petshop_2019.csv',
    'vendas_linha_petshop_2020.csv',
    'vendas_linha_petshop_2021.csv',
    'vendas_linha_petshop_2022.csv'
]

dfs = []

for arq in arquivos:
    df = pd.read_csv(arq, sep=';', decimal=',', encoding='latin1', on_bad_lines='skip')
    dfs.append(df)

# concatenar
vendas_totais = pd.concat(dfs, ignore_index=True)

# visualizar
vendas_totais.head()

,cod_pedido,regiao_pais,produto,valor,quantidade,valor_total_bruto,data,estado,formapagto,centro_distribuicao,responsavelpedido,valor_comissao,lucro_liquido,categoriaprod
0,1,Norte,Biscoito True Champion 300g,22.0,2.0,44.0,13/06/2019,Acre,Dinheiro,Rapid Pink,Carlos,0.44,17.6,Alimentação
1,2,Norte,Biscoito True Champion 300g,21.0,2.0,42.0,03/01/2019,Amapá,Pix,Rapid Pink,Maria Linda,0.42,16.8,Alimentação
2,3,Norte,Biscoito True Champion 300g,22.0,NaN,44.0,18/01/2019,Pará,Boleto Bancário,Rapid Pink,Julia,0.88,35.2,Alimentação
3,4,Norte,Biscoito True Champion 300g,19.0,4.0,88.0,19/08/2019,Rondônia,Boleto Bancário,Rapid Pink,Yuri,0.76,30.4,Alimentação
4,5,Norte,Biscoito True Champion 300g,22.0,2.0,44.0,18/10/2019,Tocantins,Cartão Crédito,Rapid Pink,Dolores,0.44,17.6,Alimentação


In [6]:
# correlacao
pearson_correl = vendas_totais.corr(numeric_only=True)
pearson_correl

,cod_pedido,valor,valor_total_bruto,valor_comissao,lucro_liquido
cod_pedido,1.000000,0.091924,0.001233,-0.013924,-0.024483
valor,0.091924,1.000000,0.061934,0.213175,0.191428
valor_total_bruto,0.001233,0.061934,1.000000,0.264349,0.262414
valor_comissao,-0.013924,0.213175,0.264349,1.000000,0.888694
lucro_liquido,-0.024483,0.191428,0.262414,0.888694,1.000000


A única correlação forte encontrada foi entre valor_comissao e lucro_liquido (0.888694)
Observa-se uma forte correlação positiva entre valor_comissao e lucro_liquido (r = 0.888694), indicando que aumentos na comissão estão associados ao aumento do lucro líquido.